app/src/androidTest/java/com/github/houseorganizer/houseorganizer/EditHouseholdTest.java

```diff
@@ -0,0 +1,389 @@
+        onView(withId(R.id.imageButtonRemoveUser)).perform(click());
+
+        // Get state of house
+        Map<String, Object> houseData_after = FirebaseTestsHelper.fetchHouseholdData(TEST_HOUSEHOLD_NAMES[0], db);
+        List<String> resident_after = (List<String>) houseData_after.get("residents");
+        Long num_residents_after = (Long) houseData_after.get("num_members");
+
+        // Compare states
+        assertEquals(resident_before, resident_after);
+        assertEquals(num_residents_before, num_residents_after);
+    }
+
+    @Test
+    public void removeUserFailsWithUnknownUser() throws ExecutionException, InterruptedException {
+        // Get state of house
+        Map<String, Object> houseData_before = FirebaseTestsHelper.fetchHouseholdData(TEST_HOUSEHOLD_NAMES[0], db);
+        List<String> resident_before = (List<String>) houseData_before.get("residents");
+        Long num_residents_before = (Long) houseData_before.get("num_members");
+
+        // Perform clicks
+        onView(withId(R.id.editTextRemoveUser)).perform(click(), typeText(UNKNOWN_USER), closeSoftKeyboard());
+        onView(withId(R.id.imageButtonRemoveUser)).perform(click());
+
+        // Get state of house
+        Map<String, Object> houseData_after = FirebaseTestsHelper.fetchHouseholdData(TEST_HOUSEHOLD_NAMES[0], db);
+        List<String> resident_after = (List<String>) houseData_after.get("residents");
+        Long num_residents_after = (Long) houseData_after.get("num_members");
+
+        // Compare states
+        assertEquals(resident_before, resident_after);
+        assertEquals(num_residents_before, num_residents_after);
+    }
+
+    @Test
+    public void removeUserFailsWithUserNotInHousehold() throws ExecutionException, InterruptedException {
+        // Get state of house
+        Map<String, Object> houseData_before = FirebaseTestsHelper.fetchHouseholdData(TEST_HOUSEHOLD_NAMES[0], db);
+        List<String> resident_before = (List<String>) houseData_before.get("residents");
+        Long num_residents_before = (Long) houseData_before.get("num_members");
+
+        // Perform clicks
+        onView(withId(R.id.editTextRemoveUser)).perform(click(), typeText(TEST_USERS_EMAILS[2]), closeSoftKeyboard());
+        onView(withId(R.id.imageButtonRemoveUser)).perform(click());
+
+        // Get state of house
+        Map<String, Object> houseData_after = FirebaseTestsHelper.fetchHouseholdData(TEST_HOUSEHOLD_NAMES[0], db);
+        List<String> resident_after = (List<String>) houseData_after.get("residents");
+        Long num_residents_after = (Long) houseData_after.get("num_members");
+
+        // Compare states
+        assertEquals(resident_before, resident_after);
+        assertEquals(num_residents_before, num_residents_after);
+    }
+
+    @Test
+    public void removeUserFailsWhenRemovingYourself() throws ExecutionException, InterruptedException {
```

Comment:
Why is this the current behaviour? Is it because the only person able to remove household members is the owner => the owner can't remove him/herself??

Reply:
Actually yes. I disallowed this because it would mean that we have to randomly assign someone else in the household as owner. What we can do in the future is a button to remove yourself from he household if you are a participant but no he owner i.e. you just quit the household

Reply:
Right, that would be a great feature! Imagine trying to quit a household but the owner won't kick you out :D